crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1018,4 +1087,66 @@ mod tests {
         assert!(entity.get_by_id(invalid_component_id).is_none());
         assert!(entity.get_mut_by_id(invalid_component_id).is_none());
     }
+
+    #[test]
```

Comment:
I would love to see a few tests verifying that the dynamic and static approaches result in identical worlds.

Reply:
Gave it a shot, not sure if there's a better way to compare worlds, let me know if I can improve it!